src/main/java/com/dao/impl/orm/order/OrderDAO.java

```diff
@@ -0,0 +1,52 @@
+package com.dao.impl.orm.order;
+
+import com.dao.entity.client.ClientEntity;
+import com.dao.entity.order.OrderEntity;
+import com.dao.entity.product.ProductEntity;
+import com.dao.exception.DAOException;
+import com.dao.impl.orm.OrmAbstractDAO;
+import com.dto.client.Client;
+import org.hibernate.Criteria;
+import org.hibernate.Session;
+import org.hibernate.Transaction;
+import org.hibernate.criterion.Restrictions;
+import org.springframework.stereotype.Repository;
+
+import java.time.LocalDate;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Optional;
+
+@Repository
+public class OrderDAO extends OrmAbstractDAO<OrderEntity> {
+
+    public OrderEntity saveOrUpdateEntity(OrderEntity entity) throws DAOException {
+        try(Session session = sessionFactory.openSession()){
+            Transaction transaction = session.beginTransaction();
+//            if (entity.getClientEntity().getId() == 0){
```

Comment:
Use _Cascades_ to configure how to save/update nested entities
If nested entities should be saved/updated separately - call this updates at service layer, not DAO